python/semantic_kernel/orchestration/sk_function.py

```diff
@@ -422,10 +483,15 @@ async def _invoke_semantic_async(self, context, settings):
                     "Semantic functions must have either an AI service or Chat service",
                 )
 
+        functions = self.describe_function_call()
+
         service = (
             self._ai_service if self._ai_service is not None else self._chat_service
         )
-        new_context = await self._function(service, settings, context)
+        if service.function_calling_enabled and functions:
```

Comment:
Lots of confusing use of the word functions 

Some suggestions:
- function_completion_enabled
- has_function_completion_api

@gitri-ms since you'll be working on this for .NET, I'd like your thoughts here as well

Reply:
If there are any naming conventions for this, please share, the openai chat completion api does use the term function_call so I went mostly from that, but getting a more generic terminology would be good, especially as other LLM's start adding similar functionality